urlac47192b66_ggrandes_bouncer_tgz-pJ8-org_javastack_bouncer_ClusterTestJ8/src/org/javastack/bouncer/Options.java

```diff
@@ -0,0 +1,348 @@
+		return getLong(name, null);
+	}
+
+	public void setLong(final String name, Long value) {
+		if (value == null) {
+			value = L_NULL;
+		}
+		longParams.put(name, value);
+	}
+
+	/**
+	 * Helper (remove options that only apply to MUX)
+	 */
+	public Options unsetOptionsMUX() {
+		unsetFlags(MUX_OUT | MUX_IN | MUX_AES | MUX_SSL);
+		return this;
+	}
+
+	/**
+	 * Helper (remove options that only apply to Plain Connections)
+	 */
+	public Options unsetOptionsPlain() {
+		unsetFlags(TUN_SSL | TUN_ENDSSL);
+		return this;
+	}
+
+	/**
+	 * Helper (remove options that only apply to Cluster Connections)
+	 */
+	public Options unsetOptionsCluster() {
+		unsetFlags(CLUSTER_AES | CLUSTER_SSL);
+		return this;
+	}
+
+	/**
+	 * Check is specified flag is active
+	 * 
+	 * @param opt
+	 * @param FLAG
+	 * @return true or false
+	 */
+	public boolean isOption(final int FLAG) {
+		return ((flags & FLAG) != 0);
+	}
+
+	/**
+	 * Return options in numeric form (bitwise-flags)
+	 * 
+	 * @param string to parse
+	 * @return int with enabled flags
+	 */
+	int parseOptions(final String str) {
+		final String[] opts = str.split(",");
+		int ret = 0;
+		for (String opt : opts) {
+			final int KEY = 0, VALUE = 1;
+			final String[] optKV = opt.split("=");
+			// Process Flags
+			final Integer f = MAP_FLAGS.get(opt.toUpperCase());
+			if (f != null) {
+				ret |= f.intValue();
+			}
+			// Process String Params
+			final String s = strParams.get(optKV[KEY].toUpperCase());
+			if (s != null) {
+				strParams.put(optKV[KEY], optKV[VALUE]);
+			}
+			// Process Integer Params
+			final Integer i = intParams.get(optKV[KEY].toUpperCase());
+			if (i != null) {
+				intParams.put(optKV[KEY], Integer.valueOf(optKV[VALUE]));
+			}
+			// Process Long Params
+			final Long l = longParams.get(optKV[KEY].toUpperCase());
+			if (l != null) {
+				longParams.put(optKV[KEY], Long.valueOf(optKV[VALUE]));
+			}
+		}
+		return ret;
+	}
+
+	/**
+	 * For humans, return options parsed/validated
+	 * 
+	 * @return human readable string
+	 */
+	@Override
+	public synchronized String toString() {
+		int i = 0;
+		final StringBuilder sb = new StringBuilder();
+		// Flags
+		for (Entry<String, Integer> e : MAP_FLAGS.entrySet()) {
+			final String key = e.getKey();
+			final Integer value = e.getValue();
+			if ((flags & value) != 0) {
+				if (i > 0)
+					sb.append(",");
+				sb.append(key);
+				i++;
+			}
+		}
+		// Strings
+		for (Entry<String, String> e : strParams.entrySet()) {
+			final String key = e.getKey();
+			final String value = e.getValue();
+			if (value != S_NULL) {
+				if (i > 0)
+					sb.append(",");
+				sb.append(key).append("=").append(value);
+				i++;
+			}
+		}
+		// Integers
+		for (Entry<String, Integer> e : intParams.entrySet()) {
+			final String key = e.getKey();
+			final Integer value = e.getValue();
+			if (value != I_NULL) {
+				if (i > 0)
+					sb.append(",");
+				sb.append(key).append("=").append(value);
+				i++;
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 287:292, 298:303.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.